pandas/core/frame.py

```diff
@@ -4621,8 +4624,14 @@ def drop_duplicates(
         if inplace:
             (inds,) = (-duplicated)._ndarray_values.nonzero()
             new_data = self._data.take(inds)
+
+            if ignore_index:
+                new_data.axes[1] = ibase.default_index(len(inds))
             self._update_inplace(new_data)
         else:
+            if ignore_index:
+                idx = ibase.default_index(len(self[-duplicated]))
```

Comment:
I think this block could be more succinctly writer as:

```python
result = self[~duplicated]
if ignore_index:
    result = result.reset_index(drop=True)

return result
```

Or something similar. FWIW I think the current method of evaluating `self[~duplicated]` twice can be costly for larger frames

Reply:
ahh, good call! @WillAyd 

will change! i think `idx = ibase.default_index(sum(-duplicated))` should be also faster than using `self[-duplicated]`